### PR TITLE
Bump Dasharo Coreboot / hotp-verification; fix nitropad-nxx ec-powerdown

### PR DESCRIPTION
**Pull request by daringer**
This is the upstream relevant part of Nitrokey's [next release](https://github.com/Nitrokey/heads/tree/nitropad-release-v2.4) for the Nitropads, it consists of three parts:

1) update Dasharo Coreboot to the most recent version (1.7.2)
  * all previous coreboot patches are now upstreamed and removed
  * added a minimal-change patch to switch available S-states based on model during compile-time
  * fixes: https://github.com/Nitrokey/heads/issues/29 (suspend compatibility improved, see below)
  * fixes: https://github.com/Nitrokey/heads/issues/25 (see tests below)
  * fixes: https://github.com/Nitrokey/heads/issues/32 (part of the dasharo coreboot bump)
  
2) fix watchdog-based powerdown for `nitropad-nxx` 
* fixes: https://github.com/Nitrokey/heads/issues/24

3) updates `hotp-verification` to obey Nitrokey 3's extended security model
  * older `hotp-verification` versions + NK3 v1.6 will not allow overwriting the HOTP secret, workaround: delete it manually using `nitropy`
  * using this `hotp-verification` version the HOTP secret is first removed before setting it again during HOTP regeneration
  * this change is fully backwards compatible with any previously working tokens
  
### Test Results

First round of tests was done for `t430-hotp-max`, `nitropad-nv41`, `nitropad-ns50`:
* QubesOS / ubuntu regular boot + installation
* splash screen visible
* flashing using the build-time-provided `.zip`
* suspend ns50 (S0ix) -> QubesOS (not ok/expected), Ubuntu (ok)
* suspend nv41 (S3) -> QubesOS (ok), Ubuntu (ok)
* igfx availability (nv41/ns50) inside Ubuntu (ok)
* igfx availability (nv41/ns50) inside QubesOS (4.1.2) still `llvm-pipe`, works with QubesOS 4.2
* branding and vendor reconfiguration works as expected (`site-local`)
* Nitrokey 3 HOTP with firmware <v1.6 & >=v1.6

We plan for another round of testing early in January, once these tests confirm the listed ones - I'll change the PR to be ready-for-review.

### CODE REVIEW

**Comment by tlaurion** [comment]
@JonathonHall-Purism time to test hotp for regression and approve?

**Comment by tlaurion** [comment]
Tested without regression
- Untested_w530-hotp-maximized (new main machine)

File: config/coreboot-nitropad-ns50.config [hunk 1]
```diff
@@ -560,7 +560,7 @@ CONFIG_CRB_TPM_BASE_ADDRESS=0xfed40000
 # CONFIG_ELOG is not set
 CONFIG_CACHE_MRC_SETTINGS=y
 CONFIG_MRC_SETTINGS_PROTECT=y
-# CONFIG_SMMSTORE is not set
+CONFIG_SMMSTORE=y
```
Comment (tlaurion) [inline config/coreboot-nitropad-ns50.config:563]: @daringer Will you remove those?

File: config/coreboot-nitropad-nv41.config [hunk 2]
```diff
@@ -560,7 +560,7 @@ CONFIG_CRB_TPM_BASE_ADDRESS=0xfed40000
 # CONFIG_ELOG is not set
 CONFIG_CACHE_MRC_SETTINGS=y
 CONFIG_MRC_SETTINGS_PROTECT=y
-# CONFIG_SMMSTORE is not set
+CONFIG_SMMSTORE=y
```
Comment (tlaurion) [inline config/coreboot-nitropad-nv41.config:563]: @daringer Will you remove those?

File: modules/hotp-verification [hunk 3]
```diff
@@ -2,12 +2,12 @@ modules-$(CONFIG_HOTPKEY) += hotp-verification
 
 hotp-verification_depends := libusb $(musl_dep)
 
-# v1.4
-hotp-verification_version := b69bb20119d3cea5ec5c13d11b213dd80dfd8334
+# v1.5
+hotp-verification_version := 70c04f51387eee8f777e943ba83b6405764a3cd2
```
Comment (tlaurion) [inline modules/hotp-verification:6]: No regression observed on Nitrokey Pro storage 2 used for HOTP and factory reset.

**Review by tlaurion — CHANGES_REQUESTED** [review]

**Comment by daringer** [comment]
At this point we cannot get rid of the `SMMSTORE` - Dasharo introduces reading from it [here](https://github.com/Dasharo/coreboot/blob/novacustom_nv4x_adl_v1.7.2/src/vendorcode/dasharo/options.c#L20):  This means that currently any Dasharo build is implicitly depending on `SMMSTORE` (even if it's not explicitly set in the configs, it will be pulled in via EDK2). Removing this might be possible, but is quite a bigger changeset which we would have to introduce and I cannot oversee currently if this would introduce regressions of some kind.

**Comment by daringer** [comment]
We also did another full test iteration today, fully confirmed what is written in "Test Results" in the initial post

**Review by JonathonHall-Purism — APPROVED** [review]
Tested Librem 14 + Librem Key, looks good to me.  Thanks @daringer

**Review by tlaurion — APPROVED** [review]

**Pull request merged**